### PR TITLE
Bug/inba 917 jf fix for login

### PR DESCRIPTION
**Pull request by jmfadeley**
#### What does this PR do?
This is a refactor of the login page. There was some bloat and some features we didn't really want or need in there. The PR was trying to remove the "click twice" issue where sometimes the login would stall. I think it succeeds.

#### Related JIRA tickets:
INBA-917.

#### How should this be manually tested?
This is a total rewrite so we have to check all functionality for login:
1) As soon as you load the page, remove any emails saved to the fields and reload. 
2) Upon reload, with blank fields, immediately hit "Sign In". The fields should turn red and tell the user that they are required.
3) Refresh the page.
4) Click on the "Email" field then click away. The field should turn red, and display "Email required"
5) Click on the "Password" field then click away. The field should turn red, and display "Password required" 
6) Fill in email and confirm that the red goes away around its field.
7) Fill in password and confirm that the red goes away around its field.
8) Fill in username and password. Click "Sign In". Confirm you sign in.
9) Log out. Repeat the login, but this time hit the "Enter" key. Confirm you sign in.
10) Logout. With a blank email field, click on "Forgot password?" Confirm no toast is displayed. Instead, the user should be told to fill in their email in the field above for the modal.
11) Fill in the email and click on "Forgot Password?" Try canceling.
12) Click on "Forgot Password?" and confirm a reset. Confirm the message about sending an email is set on the screen and not by a toast. Confirm you receive the email.
13) Click the "Remember me?" checkbox and sign in. Make sure your amida-auth-microservice has `.env` setting `AUTH_SERVICE_REFRESH_TOKEN_ENABLED=true` or this will not work (it may have been just `REFRESH_TOKEN_ENABLED` before). Then walk away for an hour or so and check back. You should still be logged in. 

Then, after receiving the email, check the following for the reset password screen:
1) Upon load, with blank fields, immediately hit "Reset password". The fields should turn red and tell the user that they are required.
2) Refresh the page.
4) Click on the "New Password" field then click away. The field should turn red, and display "Password required"
5) Click on the "Confirm Password" field then click away. The field should turn red, and display "Password confirmation required" 
6) Fill in password and confirm that the red goes away around its field.
7) Fill in confirm password and confirm that the red goes away around its field.
8) Fill in two passwords that don't match and hit "Reset Password" to receive warning about mismatched passwords.
9) Fill two passwords that match. Turn OFF amida-auth-microservice. Submit. Confirm you receive a failed message.
10) Turn on amida-auth-microservice. Submit with two matching passwords. Confirm it goes through and you are redirected. No toasts should appear.

I actually cannot remember if that last point still works or not. If not, make it another ticket.

Reedy, please check IE. Riley, feel free to weigh in on UI but understand I may draw the line if it's a lot of work.

### CODE REVIEW

**Comment by rileyorr** [comment]
@jmfadeley When I click on the "Reset Password" button in the reset pw modal the modal does not close and I don't receive an email. I also can't log in now...

**Comment by rileyorr** [comment]
@jmfadeley scratch that - got it to work thanks to @morleegan! 

My only UI change would be on the reset password page sent through the email. The first adjustment would be to provide spacing between the text input and the button so they aren't touching (I think the same spacing as on the login page would be perfect). The second adjustment would be to add a second text input to confirm password. We wouldn't want a user to have only one instance to type in their new password and accidentally type it incorrectly just to have to go through the reset password process again! 
screen shot 2019-02-22 at 10 47 10 am

**Comment by jmfadeley** [comment]
@rileyorr yeaaaah that's kind of out of scope. I can slip in the styling but it'll add two to three hours to change the reset password form to reflect these changes.

**Comment by jmfadeley** [comment]
@rileyorr screw it. Give me a minute and I'll do it.

**Comment by jmfadeley** [comment]
@rileyorr okay, try again and look over the reset password page. Updating PR for changes.

**Comment by jmfadeley** [comment]
@Arghast @eawest2 @rileyorr updated to include reset password screen.

File: src/styles/login/_login-form.scss [hunk 1]
```diff
@@ -4,16 +4,18 @@ $block-class:'login-form';
...
+        flex-direction: column;
 
         &__input {
             @include text-input;
 
-            width: 100%;
+            margin-bottom: 16px;
+            width: 384px;
+
+            &--flag {
```
Comment (morleegan) [inline src/styles/login/_login-form.scss]: add `&--flag { border: etc }` class in `/common/mixin.scss` `text-input`.  

then we don't need it here and in input

File: src/styles/resetpassword/_reset-password-form.scss [hunk 2]
```diff
@@ -13,12 +13,22 @@ $block-class: 'reset-password-form';
...
         }
 
-        &__password {
+        &__input {
             @include text-input;
+
+            width: 384px;
+            margin-bottom: 16px;
+
+            &--flag {
```
Comment (morleegan) [inline src/styles/resetpassword/_reset-password-form.scss]: we can get rid of it here if you do what is above.

**Comment by Arghast** [comment]
@jmfadeley Functionality looks good other than two minor issues. When I enter a wrong username or password it deletes strings from the "email" field which could make repeated login attempts annoying. Also, just a note that when I log out the email field is still filled in which is interesting considering it disappears when trying to log in.

**Comment by jmfadeley** [comment]
@morleegan @Arghast I am so happy... I got to use the phrase, "mixins and fixins" in the same sentence. Take a look now.

**Review by Arghast — APPROVED** [review]
@jmfadeley @morleegan LGTM! Checked on all major browsers

**Pull request merged**